aws-predictions/src/main/java/com/amplifyframework/predictions/aws/http/LivenessWebSocket.kt

```diff
@@ -162,9 +152,20 @@ internal class LivenessWebSocket(
                         if (response.challengeEvent != null) {
                             challengeType = response.challengeEvent.challengeType
                         } else if (response.serverSessionInformationEvent != null) {
+
+                            val clientRequestedOldLightChallenge = clientSessionInformation.challengeVersions
+                                .any { it == Challenge.FaceMovementAndLightChallenge("1.0.0") }
+
+                            if (challengeType == null && clientRequestedOldLightChallenge) {
+                                // For the  1.0.0 version of FaceMovementAndLight challenge, backend doesn't send a
+                                // ChallengeEvent so we need to manually check and set it if that specific challenge
+                                // was requested.
+                                challengeType = FaceLivenessChallengeType.FaceMovementAndLightChallenge
+                            }
+
                             // If challengeType hasn't been initialized by this point it's because server sent an
                             // unsupported challenge type so return an error to the client.
-                            if (!this@LivenessWebSocket::challengeType.isInitialized) {
+                            if (challengeType == null) {
                                 onErrorReceived.accept(
```

Comment:
Sorry I'm just now noticing this but don't we need to kill the websocket at this point?